pkg/trace/writer/trace_test.go

```diff
@@ -27,20 +28,28 @@ func TestTraceWriter(t *testing.T) {
 		// Create a trace writer, its incoming channel and the endpoint that receives the payloads
 		traceWriter, traceChannel, testEndpoint, _ := testTraceWriter()
 		// Set a maximum of 4 spans per payload
-		traceWriter.conf.MaxSpansPerPayload = 4
 		traceWriter.Start()
 
+		tracePkg := randomTracePackage(1, 1)
+		size := calculateTracePayloadEstimatedSize([]*TracePackage{tracePkg})
+		defer func(old int) {
+			payloadFlushThreshold = old // reset original setting
+		}(payloadFlushThreshold)
+		payloadFlushThreshold = int(size + size + 1)
+
 		// Send a few sampled traces through the writer
 		sampledTraces := []*TracePackage{
-			// These 2 should be grouped together in a single payload
-			randomTracePackage(1, 1),
-			randomTracePackage(1, 1),
-			// This one should be on its own in a single payload
-			randomTracePackage(3, 1),
-			// This one should be on its own in a single payload
+			// these two will no trigger a flush, because they are
+			// below the size threshold.
+			tracePkg,
+			tracePkg,
+			// this one will trigger a flush of the previous two,
+			// and possibly of itself.
```

Comment:
Nit: `and possible of itself` kinda feels like the test is non-deterministic. In fact, it should always be flushed by itself for the test to work.

May I suggest: `this one will trigger a flush containing the previous two packages and another flush with just this package (either immediately or when the next big package arrives)`.

The comment on line 68 also needs updating.

Reply:
You're right, this is terrible phrasing. I fixed it in the new commit.